src/pages/workspaces/workspace/Data.js

```diff
@@ -655,6 +656,7 @@ const WorkspaceData = _.flow(
     const asyncImportJobs = useStore(asyncImportJobStore);
 
     const entityServiceDataTableProvider = new EntityServiceDataTableProvider(namespace, name);
+    const region = isAzureWorkspace ? storageDetails.azureContainerRegion : storageDetails.googleBucketLocation;
```

Comment:
You should be careful about this during the period of time when the workspace details are still loading. Are you guaranteed that at this point in the code the workspace details (including all region callbacks) have completed?

For an example of what the WorkspaceDashboard does, see [Azure case](https://github.com/DataBiosphere/terra-ui/blob/f4e0f405b84cde8be674caaa8190d77d56243395/src/pages/workspaces/workspace/Dashboard.js#L527)

Reply:
Hmm, very hard to guarantee these things.  I've added a ? after `storageDetails` here - that will result in '? Unknown region' being displayed if `storageDetails` isn't available, instead of breaking.

Reply:
@calypsomatic `storageDetails` will always be defined. However, `storageDetails.azureContainerRegion` will not be undefined unless we have gotten back the Ajax response (which is why its TS definition indicates that it may not be undefined). So instead of what you suggested you need to do something like the WorkspaceDashboard example I referenced.

Reply:
Sorry about the misunderstanding!  I've updated it so to display 'Loading' with a blank flag if the region is undefined - I wasn't able to get a picture because it was too fast but I did see it display for a moment!